src/Config/Validator/Deploy/PhpVersion.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\Config\Validator\Deploy;
+
+use Composer\Composer;
+use Composer\Package\Version\VersionParser;
+use Composer\Semver\Constraint\ConstraintInterface;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Package\MagentoVersion;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Checks the current PHP version in accordance with
+ * the maximum allowed constraint of PHP of the magento/magento2-base package
+ */
+class PhpVersion implements ValidatorInterface
+{
+    /**
+     * @var Validator\ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @var Composer
+     */
+    private $composer;
+
+    /**
+     * @var VersionParser
+     */
+    private $versionParser;
+
+    /**
+     * @var MagentoVersion
+     */
+    private $magentoVersion;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param Composer $composer
+     * @param Validator\ResultFactory $resultFactory
+     * @param VersionParser $versionParser
+     * @param MagentoVersion $magentoVersion
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        Composer $composer,
+        Validator\ResultFactory $resultFactory,
+        VersionParser $versionParser,
+        MagentoVersion $magentoVersion,
+        LoggerInterface $logger
+    ) {
+        $this->composer = $composer;
+        $this->resultFactory = $resultFactory;
+        $this->versionParser = $versionParser;
+        $this->magentoVersion = $magentoVersion;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Validates PHP version
+     *
+     * @return Validator\ResultInterface
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        try {
+            $recommendedPhpConstraint = $this->getRecommendedPhpConstraint();
+            $currentPhpConstraint = $this->getCurrentPhpConstraint();
+
+            if (!$recommendedPhpConstraint->matches($currentPhpConstraint)) {
+                return $this->resultFactory->error(
+                    sprintf(
+                        'For Magento %s recommended PHP version satisfying the constraint %s. '
+                        . 'Currently installed PHP version %s',
+                        $this->magentoVersion->getVersion(),
+                        $recommendedPhpConstraint->getPrettyString(),
+                        $currentPhpConstraint->getPrettyString()
+                    ),
+                    "Change the version of PHP to the version that satisfies the restriction conditions.\n"
```

Comment:
"Change the PHP configuration in the magento.app.yaml file to specify a version that meets the requirement,\n"